### PR TITLE
Add retry decorator and use it.

### PR DESCRIPTION
**Pull request by jonathanmetzman**

### CODE REVIEW

File: infra/build_specified_commit.py [hunk 1]
```diff
@@ -144,17 +143,10 @@ def copy_src_from_docker(project_name, host_dir):
   return src_dir
 
 
+@retry.wrap(_IMAGE_BUILD_TRIES, 2, 'infra.build_specified_commit')
```
Comment (inferno-chromium) [inline infra/build_specified_commit.py:146]: infra.build_specified_commit. _build_image_with_retries for function name

**Review by inferno-chromium — APPROVED** [review]

File: infra/build_specified_commit.py (hunk 1 above)
Comment (jonathanmetzman) [reply infra/build_specified_commit.py]: Done.

File: infra/retry.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+def get_delay(num_try, delay, backoff):
+  """Compute backoff delay."""
+  return delay * (backoff**(num_try - 1))
+
+
+def wrap(retries,
+         delay,
+         function,
```
Comment (oliverchang) [inline infra/retry.py:38]: We can simplify this here by not requiring `function`. We needed this in CF to get the full qualified function name for metrics reporting, which is not needed here. We can use something like `__qualname__` on function to get the name (and possibly clean up CF as well in the future).

Comment (jonathanmetzman) [reply infra/retry.py:38]: Done.

**Pull request merged**